launcher/minecraft/mod/tasks/LocalModParseTask.cpp

```diff
@@ -199,6 +229,42 @@ ModDetails ReadMCModTOML(QByteArray contents)
     }
     details.icon_file = logoFile;
 
+    auto parseDep = [&details](toml::array* dependencies) {
+        if (dependencies) {
+            for (auto& dep : *dependencies) {
+                auto dep_table = dep.as_table();
+                if (dep_table) {
+                    auto modId = dep_table->get("modId")->value_or<std::string>("");
+                    if (modId != "forge" && modId != "neoforge" && modId != "minecraft") {
+                        if (dep_table->contains("type") && (dep_table->get("type"))->value_or<std::string>("") == "required") {
+                            details.dependencies.append(QString::fromStdString(modId));
+                        } else if (dep_table->contains("mandatory") && (dep_table->get("mandatory"))->value_or(false)) {
+                            details.dependencies.append(QString::fromStdString(modId));
+                        }
+                    }
+                }
+            }
+        }
```

Comment:
Oops I guess I reworked a whole function again...
Better null safety and less nesting
```suggestion
        if (!dependencies) {
            return;
        }
        for (auto& dep : *dependencies) {
            auto dep_table = dep.as_table();
            if (!dep_table) {
               continue;
            }

            auto modId = (*dep_table)["modId"].as_string();
            if (!modId) {
                continue;
            }
            if (modId->get() == "forge" || modId->get() == "neoforge" || modId->get() == "minecraft") {
                continue;
            }

            if (auto type = (*dep_table)["type"].as_string()) { // NeoForge format
                if (type->get() != "required") {
                    continue;
                }
            } else if (auto mandatory = (*dep_table)["mandatory"]) {
                if (!mandatory->get()) {
                    continue;
                }
            } else {
                continue;
            }

            details.dependencies.append(QString::fromStdString(modId->get()));
        }
```

Reply:
As much as I love the go approach here, I think it would just be better to keep a relatively smaller number of continue: use it in moderation
I applied this suggestion partially, as you would see: I consider the 3 continue at the end being a bit too much

Reply:
No null safety though for modID - it will segfault

Reply:
I also prefer the [key].as_whatever() approach as it is null safe without needing to check .contains 
(it's also used elsewhere in the same function)

Reply:
Also given the length of 
`details.dependencies.append(QString::fromStdString(modId->get()));`
it seems bad to duplicate it for every case

Reply:
Ok now it should address all the comments but without the extra continue